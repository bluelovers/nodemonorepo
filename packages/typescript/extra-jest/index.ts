import * as snap from './lib/snap'
import * as setupTeardown from './lib/setup-teardown'
import * as snapSpawn from './lib/snap-spawn'
export { snap, setupTeardown, snapSpawn }
