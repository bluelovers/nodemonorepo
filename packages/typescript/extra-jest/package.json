{
  "name": "extra-jest",
  "version": "0.0.17",
  "description": "Some extra functions to work with jest testing framework",
  "main": "index.js",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "jest"
  ],
  "dependencies": {
    "ramda": "^0.25.0",
    "fs-extra": "^6.0.1",
    "tslib": "^1.9.2",
    "fs-tree-utils": "^0.0.14",
    "unique-temp-path": "^0.0.9",
    "@types/ramda": "^0.25.32",
    "@types/fs-extra": "^5.0.2",
    "@types/node": "^10.3.0"
  },
  "peerDependencies": {
    "jest": "^23.1.0"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
