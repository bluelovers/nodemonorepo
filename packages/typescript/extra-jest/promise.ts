export function snapRejected<X> (promise: Promise<X>) {
  return promise.then(
    value => {
      throw new Error(`Expecting promise to reject but it didn't`)
    },
    reason => {
      expect(reason).toMatchSnapshot()
    }
  )
}
