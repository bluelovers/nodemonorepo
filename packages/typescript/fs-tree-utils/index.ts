import traverse from './lib/traverse'
import read from './lib/read'
import create from './lib/create'
export * from './lib/.types'
export { traverse, read, create }
