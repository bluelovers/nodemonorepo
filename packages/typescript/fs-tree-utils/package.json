{
  "name": "fs-tree-utils",
  "version": "0.0.5",
  "description": "Work with filesystem tree with ease",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "filesystem",
    "tree"
  ],
  "main": "index.js",
  "dependencies": {
    "fs-extra": "^6.0.1",
    "ramda": "^0.25.0",
    "@types/fs-extra": "^5.0.2",
    "@types/ramda": "^0.25.28",
    "@types/node": "^10.0.8"
  },
  "devDependencies": {
    "unique-temp-path": "*"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
