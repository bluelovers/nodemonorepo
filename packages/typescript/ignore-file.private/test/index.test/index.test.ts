import * as fsTreeUtils from 'fs-tree-utils'
import * as xjest from 'extra-jest'
import virtualEnvironment from '../.lib/virtual-env'
import * as subject from '../../index'

const { apply } = virtualEnvironment
const snap = (x: any) => xjest.snap.safe(x)()

it('add', () => {
  expect(subject.add(
    ['middleA', 'removeA', 'middleB', 'removeB'],
    {
      prepend: ['headA', 'headB'],
      append: ['tailA', 'tailB'],
      remove: ['removeA', 'removeB']
    }
  )).toMatchSnapshot()
})

it('getIgnoreFiles', apply(async () => {
  expect(await subject.getIgnoreFiles('.myignore')).toMatchSnapshot()
}))

it('getFilePairs', apply(async () => {
  snap(await subject.getDeltaPairs('.myignore'))
}))

it('getDeltaPairs', apply(async () => {
  const pairs = await subject.getDeltaPairs('.myignore')
  snap(pairs)
  snap(pairs.map(({ delta, ignore }) => ({
    ignore,
    delta: delta.filter(x => Object.keys(x.content).length !== 0)
  })))
}))

it('writeIgnoreFiles', apply(async () => {
  await subject.writeIgnoreFiles(
    'root/.myignore',
    '.myignore',
    'root/container/**'
  )

  snap(await fsTreeUtils.read.nested('root'))

  snap(
    Object
      .entries((await fsTreeUtils.read.flat('root')).fileContents)
      .map(([name, content]) => ({ name, content }))
      .filter(x => /\.myignore$/.test(x.name))
      .map(({ name, content }) => [{ name }, { content }])
  )
}))

describe('minor features', () => {
  const sample = [
    'foo',
    'foo/bar',
    'foo/bar/baz'
  ]

  it('createFileChooser.byExt', () => {
    const ext = ['', ...'abc', 'abc']
    const choose = subject.createFileChooser.byExt(...ext)

    expect(
      sample.map(base => [{ base, ext }, choose(base)])
    ).toMatchSnapshot()
  })

  it('DEFAULT_FILE_CHOOSER', () => {
    expect(
      sample.map(base => [{ base }, subject.DEFAULT_DELTA_FILES_CHOOSER(base)])
    ).toMatchSnapshot()
  })
})
