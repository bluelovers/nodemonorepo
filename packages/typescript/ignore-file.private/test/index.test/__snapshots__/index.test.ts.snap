// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add 1`] = `
Array [
  "headA",
  "headB",
  "middleA",
  "middleB",
  "tailA",
  "tailB",
]
`;

exports[`getDeltaPairs 1`] = `
"
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
  ignore: root/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
  ignore: root/container/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileEverywhere/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/bar/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileEverywhere/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/foo/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/bar/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/foo/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileInSubfolderOnly/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/bar/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileInSubfolderOnly/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/foo/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/bar/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/foo/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/foo/.myignore

"
`;

exports[`getDeltaPairs 2`] = `
"
- delta: []
  ignore: root/.myignore
- delta: []
  ignore: root/container/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
  ignore: root/container/withDeltaFileEverywhere/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileEverywhere/bar/.myignore.yaml
  ignore: root/container/withDeltaFileEverywhere/bar/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileEverywhere/foo/.myignore.yaml
  ignore: root/container/withDeltaFileEverywhere/foo/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
  ignore: root/container/withDeltaFileInContainerOnly/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
  ignore: root/container/withDeltaFileInContainerOnly/bar/.myignore
- delta:
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
  ignore: root/container/withDeltaFileInContainerOnly/foo/.myignore
- delta: []
  ignore: root/container/withDeltaFileInSubfolderOnly/.myignore
- delta:
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileInSubfolderOnly/bar/.myignore.yaml
  ignore: root/container/withDeltaFileInSubfolderOnly/bar/.myignore
- delta:
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileInSubfolderOnly/foo/.myignore.yaml
  ignore: root/container/withDeltaFileInSubfolderOnly/foo/.myignore
- delta: []
  ignore: root/container/withoutAnyDeltaFile/.myignore
- delta: []
  ignore: root/container/withoutAnyDeltaFile/bar/.myignore
- delta: []
  ignore: root/container/withoutAnyDeltaFile/foo/.myignore

"
`;

exports[`getFilePairs 1`] = `
"
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
  ignore: root/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
  ignore: root/container/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileEverywhere/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/bar/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileEverywhere/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileEverywhere/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileEverywhere/foo/.myignore.yml
  ignore: root/container/withDeltaFileEverywhere/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/bar/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content:
        append:
          - tailContainer123
          - tailContainer456
        prepend:
          - headContainer123
          - headContainer456
        remove:
          - notInContainer
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInContainerOnly/foo/.myignore.yml
  ignore: root/container/withDeltaFileInContainerOnly/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInBar
      filename: root/container/withDeltaFileInSubfolderOnly/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/bar/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yaml
    - content:
        append:
          - tailFoo123
          - tailFoo456
        prepend:
          - headFoo123
          - headFoo456
        remove:
          - notInFoo
      filename: root/container/withDeltaFileInSubfolderOnly/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/.myignore.yml
    - content: {}
      filename: root/container/withDeltaFileInSubfolderOnly/foo/.myignore.yml
  ignore: root/container/withDeltaFileInSubfolderOnly/foo/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/bar/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/bar/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/bar/.myignore
- delta:
    - content: {}
      filename: root/.myignore.yaml
    - content: {}
      filename: root/container/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yaml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/foo/.myignore.yaml
    - content: {}
      filename: root/.myignore.yml
    - content: {}
      filename: root/container/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/.myignore.yml
    - content: {}
      filename: root/container/withoutAnyDeltaFile/foo/.myignore.yml
  ignore: root/container/withoutAnyDeltaFile/foo/.myignore

"
`;

exports[`getIgnoreFiles 1`] = `
Array [
  "root/.myignore",
  "root/container/.myignore",
  "root/container/withDeltaFileEverywhere/.myignore",
  "root/container/withDeltaFileEverywhere/bar/.myignore",
  "root/container/withDeltaFileEverywhere/foo/.myignore",
  "root/container/withDeltaFileInContainerOnly/.myignore",
  "root/container/withDeltaFileInContainerOnly/bar/.myignore",
  "root/container/withDeltaFileInContainerOnly/foo/.myignore",
  "root/container/withDeltaFileInSubfolderOnly/.myignore",
  "root/container/withDeltaFileInSubfolderOnly/bar/.myignore",
  "root/container/withDeltaFileInSubfolderOnly/foo/.myignore",
  "root/container/withoutAnyDeltaFile/.myignore",
  "root/container/withoutAnyDeltaFile/bar/.myignore",
  "root/container/withoutAnyDeltaFile/foo/.myignore",
]
`;

exports[`writeIgnoreFiles 1`] = `
"
.myignore: |

  # to be included
  root123
  root456

  # some are eliminated
  notInContainer
  notInFoo
  notInBar

  # in addition, comments are eliminated
container:
  withDeltaFileEverywhere:
    .myignore: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
    .myignore.yaml: |
      prepend:
        - headContainer123
        - headContainer456
      append:
        - tailContainer123
        - tailContainer456
      remove:
        - notInContainer
    bar:
      .myignore: |-
        headFoo123
        headFoo456
        headContainer123
        headContainer456
        root123
        root456
        notInFoo
        tailContainer123
        tailContainer456
        tailFoo123
        tailFoo456
      .myignore.yaml: |
        prepend:
          - headFoo123
          - headFoo456
        append:
          - tailFoo123
          - tailFoo456
        remove:
          - notInBar
    foo:
      .myignore: |-
        headFoo123
        headFoo456
        headContainer123
        headContainer456
        root123
        root456
        notInBar
        tailContainer123
        tailContainer456
        tailFoo123
        tailFoo456
      .myignore.yaml: |
        prepend:
          - headFoo123
          - headFoo456
        append:
          - tailFoo123
          - tailFoo456
        remove:
          - notInFoo
  withDeltaFileInContainerOnly:
    .myignore: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
    .myignore.yaml: |
      prepend:
        - headContainer123
        - headContainer456
      append:
        - tailContainer123
        - tailContainer456
      remove:
        - notInContainer
    bar:
      .myignore: |-
        headContainer123
        headContainer456
        root123
        root456
        notInFoo
        notInBar
        tailContainer123
        tailContainer456
    foo:
      .myignore: |-
        headContainer123
        headContainer456
        root123
        root456
        notInFoo
        notInBar
        tailContainer123
        tailContainer456
  withDeltaFileInSubfolderOnly:
    .myignore: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar
    bar:
      .myignore: |-
        headFoo123
        headFoo456
        root123
        root456
        notInContainer
        notInFoo
        tailFoo123
        tailFoo456
      .myignore.yaml: |
        prepend:
          - headFoo123
          - headFoo456
        append:
          - tailFoo123
          - tailFoo456
        remove:
          - notInBar
    foo:
      .myignore: |-
        headFoo123
        headFoo456
        root123
        root456
        notInContainer
        notInBar
        tailFoo123
        tailFoo456
      .myignore.yaml: |
        prepend:
          - headFoo123
          - headFoo456
        append:
          - tailFoo123
          - tailFoo456
        remove:
          - notInFoo
  withoutAnyDeltaFile:
    .myignore: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar
    bar:
      .myignore: |-
        root123
        root456
        notInContainer
        notInFoo
        notInBar
    foo:
      .myignore: |-
        root123
        root456
        notInContainer
        notInFoo
        notInBar

"
`;

exports[`writeIgnoreFiles 2`] = `
"
- - name: root/.myignore
  - content: |

      # to be included
      root123
      root456

      # some are eliminated
      notInContainer
      notInFoo
      notInBar

      # in addition, comments are eliminated
- - name: root/container/withDeltaFileEverywhere/.myignore
  - content: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
- - name: root/container/withDeltaFileEverywhere/bar/.myignore
  - content: |-
      headFoo123
      headFoo456
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      tailContainer123
      tailContainer456
      tailFoo123
      tailFoo456
- - name: root/container/withDeltaFileEverywhere/foo/.myignore
  - content: |-
      headFoo123
      headFoo456
      headContainer123
      headContainer456
      root123
      root456
      notInBar
      tailContainer123
      tailContainer456
      tailFoo123
      tailFoo456
- - name: root/container/withDeltaFileInContainerOnly/.myignore
  - content: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
- - name: root/container/withDeltaFileInContainerOnly/bar/.myignore
  - content: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
- - name: root/container/withDeltaFileInContainerOnly/foo/.myignore
  - content: |-
      headContainer123
      headContainer456
      root123
      root456
      notInFoo
      notInBar
      tailContainer123
      tailContainer456
- - name: root/container/withDeltaFileInSubfolderOnly/.myignore
  - content: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar
- - name: root/container/withDeltaFileInSubfolderOnly/bar/.myignore
  - content: |-
      headFoo123
      headFoo456
      root123
      root456
      notInContainer
      notInFoo
      tailFoo123
      tailFoo456
- - name: root/container/withDeltaFileInSubfolderOnly/foo/.myignore
  - content: |-
      headFoo123
      headFoo456
      root123
      root456
      notInContainer
      notInBar
      tailFoo123
      tailFoo456
- - name: root/container/withoutAnyDeltaFile/.myignore
  - content: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar
- - name: root/container/withoutAnyDeltaFile/bar/.myignore
  - content: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar
- - name: root/container/withoutAnyDeltaFile/foo/.myignore
  - content: |-
      root123
      root456
      notInContainer
      notInFoo
      notInBar

"
`;
